### PR TITLE
[DS-173] Improve Mnemonic Reader's Preprocessor 

### PR DESCRIPTION
**Pull request by happilyeverafter95**
## PR Goals and Notes for Reviewers: 

Mnemonic Reader preprocessor was initially dividing Kiite Google Drives by files and finding the file(s) most likely to contain the answer. Usually, the prediction time for this would be 1-2 seconds, but in the event that the file(s) were very large (we have files close to 10k words), the prediction could take over a minute. In an effort to reduce latency and keep prediction times consistent, the goal of this PR is to restructure the preprocessor. 

The restructured preprocessor would do the following: 

* find the top x candidate file(s) likely to contain the answer 
* out of the top x file(s), find the top y paragraph(s) likely to contain the answer
* use Mnemonic Reader to make predictions over the top y paragraph(s)

## Changes Proposed in this PR: 

* Separated preprocessor from prediction function (processQuestion) as its own function (preprocess) 
* Preprocessing step is ran twice: first to find the top x files, then to find the top y paragraphs amongst the top x files 
* Prediction function (processQuestion) takes in an additional parameter to determine both the number of top files and top paragraphs to take

### CODE REVIEW

File: script/interactive.py [hunk 1]
```diff
@@ -83,7 +100,7 @@ def returnSentence(document, answer):
         if answer in sentences[i]: 
             return sentences[i]
 
-    # if answer spreads over two sentences (rare) 
+    # if answer spreads over two sentences (rare), return both sentences
```
Comment (korger) [inline script/interactive.py]: Is it possible to substantiate the claim that it's rare that an answer spreads over two sentences?  The answers that our implementation team creates usually comprise of many sentences.  Or is this just what MR considers to be an answer, so you mean that it's rare for MR to return more than one sentence, not that the answer itself can't span over more sentences?

Comment (happilyeverafter95) [reply script/interactive.py]: Yes, you are right. It's rare for MR to return an answer that spans more than one sentence. Answers from MR's training set are usually just a few words in length. Out of the MR answers that I have encountered, only a few spanned more than one sentence. This doesn't mean that it's rare for the actual answer to span over one sentence.

File: script/interactive.py [hunk 2]
```diff
@@ -206,30 +234,33 @@ def process(document, question,first, candidates=None, top_n=1):
...
         correctParagraph = questions[i][2]
         ignore = questions[i][1]
-        result = processQuestion(questions[i][0], doc, num)[0]
+        result = processQuestion(questions[i][0], doc, n1, n2)[0]
         topParagraphs = [i[0] for i in result]
-        if correctParagraph in topParagraphs[:num]:
+        if correctParagraph in topParagraphs[:n1]:
             naiveScore += 1
-        confidence = int(result[0][1])
+        confidence = result[0][1][0][0]
```
Comment (korger) [inline script/interactive.py]: Would it be possible to use some descriptive notation for indexing?  It's hard to follow what the indices [0][1][0][0] mean or how result is structured.  No action is needed for now, but please consider this for later changes!
